server/src/main/java/org/opensearch/index/query/NestedQueryBuilder.java

```diff
@@ -401,16 +401,32 @@ protected void doBuild(SearchContext parentSearchContext, InnerHitsContext inner
                 }
             }
             String name = innerHitBuilder.getName() != null ? innerHitBuilder.getName() : nestedObjectMapper.fullPath();
-            ObjectMapper parentObjectMapper = queryShardContext.nestedScope().nextLevel(nestedObjectMapper);
-            NestedInnerHitSubContext nestedInnerHits = new NestedInnerHitSubContext(
-                name,
-                parentSearchContext,
-                parentObjectMapper,
-                nestedObjectMapper
-            );
-            setupInnerHitsContext(queryShardContext, nestedInnerHits);
-            queryShardContext.nestedScope().previousLevel();
-            innerHitsContext.addInnerHitDefinition(nestedInnerHits);
+            ObjectMapper parentObjectMapper = queryShardContext.nestedScope().getObjectMapper();
```

Comment:
So, to make sure I'm understanding properly, this is returning the current top of the scope stack (without removing it). Meanwhile, the call to `nextLevel` has been moved down to line 414, which pushes `nestedObjectMapper` onto the nested scope stack. So, the previous behavior around `parentObjectMapper` is not changed.

The functional change is that we (optionally) use `parentObjectMapper.nestedTypeFilter()` to load a cached `parentFilter`, which we inject into the `queryShardContext`. In the `finally` block, we undo that change, putting the previously-scoped `parentFilter` back into the `queryShardContext`.

@heemin32 -- did I get all of that right? Assuming I did, the change makes sense to me.

Reply:
That is a correct understanding.